[CK_TILE] moe_sorting support "local_tokens" feature for EP case (#2335)
ROCm/composable_kernel: example/ck_tile/13_moe_sorting/README.md

## example
```
args:

>            -v    weather do CPU validation or not (default:1)

Reviewer: ```suggestion
           -v    turn CPU validation on (1) or off (0). Default is on (1).
```
I'm assuming it's 1 or 0.

BTW, if you don't like this wording and want to keep the original, it should be "whether" and not "weather" :)

>         -pr_i    index data type. (currently only int32 supported now) (default:int32)

Reviewer: ```suggestion
        -pr_i    index data type.  Only int32 is currently supported (default:int32)
```
```

>         -pr_w    output weight data type(currently only fp32 supported now) (default:fp32)

Reviewer: ```suggestion
        -pr_w    output weight data type. Only fp32 is currently supported (default:fp32)
```

           -t    number of input tokens (default:128)

>      -local_t    number of local input tokens, dynamic token feature,  (default:-1)

Reviewer: This sounds like it takes two arguments rather than one.

Can you give some context to clarify this?

Author: local input tokens, is needed for EP case, where the token is inside a GPU buffer, not a scalar. Let me update descriptiuon

                 will be used for EP case where each rank have different tokens.

>                  This value will be stored in GPU buffer for cuda graph usage. if -1, then no this value/buffer

Reviewer: This isn't really clear, even just from a gramatical point of view. Can you provide some details about this so we can make this clearer?

Author: I think it is clear enough, the number of token come from a GPU buffer instead of a scalar, this is the only thing needed inside this PR

>            -e    number of num_experts (default:8)

Reviewer: num_experts needs to be defined or explained

Author: I think you need understand what is the algorithm of Moe. To explain everyhing in such sorting kernel (which is utility) does not make sense

           -k    topk (default:4)
        -unit    unit_size (default:32)
-moe_buf_size    moe_buf_size (default:0)

>    -local_eid    a list of experts enabled as local expert. e.g. "0,1,4,5" (default:-1)

Reviewer: ```suggestion
   -local_eid   comma-separated list of experts enabled as local expert. The list must be in ascending order and enclosed in quotation marks. For example, `"0,1,4,5"` (default:-1)
```
does it need to be in quotes? I assumed it did, but if it doesn't, remove the part about quotation marks, and remove the actual quotation marks.

Author: it can be quote or not, depends on the shell syntax. It's the users responsibility to pass proper syntax to this, follow the shell

>                  please make sure eid is in ascending order!

Reviewer: can be removed. covered in the previous line

Author: this is to hint shell syntax

>         -seed    seed to be used, -1 means random every time (default:-1)

Reviewer: ```suggestion
        -seed    seed to be used. When set to -1, a random seed is used on each iteration (default:-1)
```
I assume it's each iteration.

Author: -1 means generate a random seed each time invoking this example. Otherwise use that seed passed by used ( to fix the randomness of this program, easy to debug)

>        -kname    when set to 1 it will print kernel name (default:0)

Reviewer: ```suggestion
       -kname    prints the kernel name when set to 1 (default:0)
```
